aten/src/ATen/native/native_functions.yaml

```diff
@@ -1484,6 +1484,11 @@
   variants: function, method
   device_guard: False
 
+- func: isreal(Tensor self) -> Tensor
+  use_c10_dispatcher: full
+  variants: function, method
+  device_guard: False
```

Comment:
Why set the device guard to false? Do the at::ones_like and at::imag calls set the device guard?

Reply:
This was just a copy-paste. I just checked and at::ones_like doesn't set device_guard and at::imag sets device_guard to False. What determines setting it to False?

Reply:
I'm not totally sure. In some special cases it may be OK, but I would just remove the "device_guard" line for safety.